tasks/build/install-k3s.yml

```diff
@@ -17,9 +17,17 @@
   notify:
     - reload systemd
   become: "{{ k3s_become_for_systemd | ternary(true, false, k3s_become_for_all) }}"
+  register: k3s_service_unit
 
 - meta: flush_handlers
 
+- name: Ensure k3s restarts after reload systemd, and if service unit file change on control plane
+  debug:
+    msg: Notify restart k3s
+  notify:
+    - restart k3s
+  when: k3s_service_unit.changed  # noqa 503
```

Comment:
Yes, this is a bit hacky. Still having a bit of a look and I might be able to come up with something, though it does seem to pass testing.

Reply:
Thanks for taking a look.  I'm open to ideas.